### PR TITLE
Do not include dependency on Standard library

### LINKED ISSUE #741: Autosubstitution depends on packages pulled in by default by the Microsoft.Quantum.Sdk
## Describe the bug
To keep repo dependencies clean, no packages in qsharp-runtime can have dependencies on QuantumLibraries, as the dependencies need to flow just one way. 

By default, though, the Microsoft.Quantum.Sdk adds dependencies to Standard unless turned off by the 
`IncludeQSharpCorePackages` property, so this property is required in all projects on the qsharp-runtime repo.

## To Reproduce
Check the package dependencies in either VS or nuget, and check that Standard is marked as dependency.

### PR DESCRIPTION
**Pull request by anpaz**
Fixes #741

### CODE REVIEW

**Comment by guenp** [comment]
@anpaz @msoeken should we merge this PR first and then resolve naming conflicts in this PR? https://github.com/microsoft/qsharp-runtime/pull/743

File: src/Simulation/AutoSubstitution/AutoSubstitution.csproj [hunk 1]
```diff
@@ -11,11 +11,22 @@
...
+    <QSharpDocsGeneration>false</QSharpDocsGeneration>
+  </PropertyGroup>
+
   <ItemGroup>
     <PackageReference Include="Microsoft.Quantum.Compiler" Version="0.17.2106147878-beta" />
     <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="3.6.0" />
   </ItemGroup>
 
+  <ItemGroup>
+    <ProjectReference Include="..\CSharpGeneration\Microsoft.Quantum.CSharpGeneration.fsproj" />
```
Comment (bettinaheim) [inline src/Simulation/AutoSubstitution/AutoSubstitution.csproj:26]: I believe this project also relies on that C# code for is generated.

File: src/Simulation/AutoSubstitution/AutoSubstitution.csproj [hunk 2]
```diff
@@ -11,11 +11,22 @@
...
+    <ProjectReference Include="..\QSharpFoundation\Microsoft.Quantum.QSharp.Foundation.csproj" />
+  </ItemGroup>
+
   <ItemGroup>
```
Comment (anpaz) [inline src/Simulation/AutoSubstitution/AutoSubstitution.csproj:30]: do you need to add the `BeforeCSharpCompile` target as in the other project?

**Review by anpaz — COMMENTED** [review]
lgtm (I can't approve since I started the review)

**Comment by bettinaheim** [comment]
@anpaz @msoeken changes were incorporated into #748.

File: src/Simulation/AutoSubstitution/AutoSubstitution.csproj (hunk 2 above)
Comment (msoeken) [reply src/Simulation/AutoSubstitution/AutoSubstitution.csproj:30]: It did not work without adding this part.

**Comment by msoeken** [comment]
> @anpaz @msoeken changes were incorporated into #748.

@beheim Should we close this PR then?

**Comment by bettinaheim** [comment]
@msoeken I believe it can be, yes.

**Pull request closed without merging**